microprofile/microprofile-lra/participant/src/main/java/org/wildfly/extension/microprofile/lra/participant/MicroProfileLRAParticipantAdd.java

```diff
@@ -0,0 +1,108 @@
+
+import org.jboss.as.controller.AbstractBoottimeAddStepHandler;
+import org.jboss.as.controller.CapabilityServiceBuilder;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.server.AbstractDeploymentChainStep;
+import org.jboss.as.server.DeploymentProcessorTarget;
+import org.jboss.as.server.deployment.Phase;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.wildfly.clustering.service.CompositeDependency;
+import org.wildfly.clustering.service.ServiceSupplierDependency;
+import org.wildfly.clustering.service.SupplierDependency;
+import org.wildfly.extension.microprofile.lra.participant._private.MicroProfileLRAParticipantLogger;
+import org.wildfly.extension.microprofile.lra.participant.deployment.LRAParticipantDeploymentDependencyProcessor;
+import org.wildfly.extension.microprofile.lra.participant.deployment.LRAParticipantDeploymentSetupProcessor;
+import org.wildfly.extension.microprofile.lra.participant.deployment.LRAParticipantJaxrsDeploymentUnitProcessor;
+import org.wildfly.extension.microprofile.lra.participant.service.LRAParticipantService;
+import org.wildfly.extension.undertow.Capabilities;
+import org.wildfly.extension.undertow.Host;
+import org.wildfly.extension.undertow.UndertowService;
+
+import java.util.Arrays;
+
+import static org.jboss.as.controller.OperationContext.Stage.RUNTIME;
+import static org.wildfly.extension.microprofile.lra.participant.MicroProfileLRAParticipantExtension.SUBSYSTEM_NAME;
+import static org.wildfly.extension.microprofile.lra.participant.MicroProfileLRAParticipantSubsystemDefinition.ATTRIBUTES;
+
+class MicroProfileLRAParticipantAdd extends AbstractBoottimeAddStepHandler {
+    static MicroProfileLRAParticipantAdd INSTANCE = new MicroProfileLRAParticipantAdd();
+
+    private MicroProfileLRAParticipantAdd() {
+        super(Arrays.asList(ATTRIBUTES));
+    }
+
+    @Override
+    protected void populateModel(ModelNode operation, ModelNode model) throws OperationFailedException {
+        MicroProfileLRAParticipantLogger.LOGGER.trace("MicroProfileLRAParticipantAdd.populateModel");
+        MicroProfileLRAParticipantSubsystemDefinition.LRA_COORDINATOR_URL.validateAndSet(operation, model);
+        MicroProfileLRAParticipantSubsystemDefinition.PROXY_SERVER.validateAndSet(operation, model);
+        MicroProfileLRAParticipantSubsystemDefinition.PROXY_HOST.validateAndSet(operation, model);
+    }
+
+    @Override
+    protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+        super.performBoottime(context, operation, model);
+        final String url = MicroProfileLRAParticipantSubsystemDefinition.LRA_COORDINATOR_URL.resolveModelAttribute(context, model).asString();
+        System.setProperty("lra.coordinator.url", url);
```

Comment:
Is there an issue somewhere to get a different configuration API for whatever is being controlled by this property? Integrating via system props is kind of yuck as there's no control over what code writes.

Reply:
I've created https://issues.redhat.com/browse/JBTM-3763